Fireworks/Core/src/FWFileEntry.cc

```diff
@@ -98,26 +107,44 @@ void FWFileEntry::openFile(bool checkVersion)
       }
    }
 
-   // load event
-   m_event = new fwlite::Event(m_file);
-
-   if (m_event->size() == 0)
-         throw std::runtime_error("fwlite::Event size == 0");
-
-
    m_eventTree = dynamic_cast<TTree*>(m_file->Get("Events"));
 
    if (m_eventTree == 0)
    { 
       throw std::runtime_error("Cannot find TTree 'Events' in the data file");
    }
 
-   // This now set in DataHelper
-   //TTreeCache::SetLearnEntries(2);
-   //m_eventTree->SetCacheSize(10*1024*1024);
-   //TTreeCache *tc = (TTreeCache*) m_file->GetCacheRead();
-   //tc->AddBranch(m_event->auxBranch_,kTRUE);
-   //tc->StartLearningPhase();
+   // Initialize caching if we are accessing a remote file
+   //if (strcmp(m_file->ClassName(), "TNetXNGFile") == 0)
+   {
+     if (FWTTreeCache::IsLogging())
+       printf("FWFileEntry::openFile enabling FWTTreeCache for file class '%s'.", m_file->ClassName());
+
+     auto tc = new FWTTreeCache(m_eventTree, 50*1024*1024);
```

Comment:
Might be reasonable to make this a configuration file - there could be significant memory usage if many files are open.